test/e2e/aws_test.go

```diff
@@ -45,101 +47,285 @@ import (
 )
 
 var (
-	cniManifests  = capiFlag.DefineOrLookupStringFlag("cniManifests", "https://docs.projectcalico.org/v3.9/manifests/calico.yaml", "URL to CNI manifests to load")
-	kubectlBinary = capiFlag.DefineOrLookupStringFlag("kubectlBinary", "kubectl", "path to the kubectl binary")
+	cniManifests      = capiFlag.DefineOrLookupStringFlag("cniManifests", "https://docs.projectcalico.org/v3.9/manifests/calico.yaml", "URL to CNI manifests to load")
+	kubectlBinary     = capiFlag.DefineOrLookupStringFlag("kubectlBinary", "kubectl", "path to the kubectl binary")
+	availabilityZones []*string
 )
 
 const (
 	scaleUpReplicas   = 5
 	scaleDownReplicas = 0
+	privateCIDR       = "10.0.2.0/24"
+	publicCIDR        = "10.0.3.0/24"
 )
 
+type testSetup struct {
+	namespace               string
+	clusterName             string
+	awsClusterName          string
+	cpMachinePrefix         string
+	cpAWSMachinePrefix      string
+	cpBootstrapConfigPrefix string
+	mdBootstrapConfig       string
+	machineDeploymentName   string
+	awsMachineTemplateName  string
+	testTmpDir              string
+	instanceType            string
+	initialReplicas         int32
+	multipleAZ              bool
+	availabilityZone        *string
+	subnetId                *string
+}
+
 var _ = Describe("functional tests", func() {
 	var (
-		namespace               string
-		clusterName             string
-		awsClusterName          string
-		cpMachinePrefix         string
-		cpAWSMachinePrefix      string
-		cpBootstrapConfigPrefix string
-		mdBootstrapConfig       string
-		machineDeploymentName   string
-		awsMachineTemplateName  string
-		testTmpDir              string
-		initialReplicas         int32
```

Comment:
I'd avoid this change for now to avoid potential merge conflicts with other PRs for now, it might also be good to separate this type of change as an independent PR.

Reply:
Will do.